### PR TITLE
Integrating gps

### PR DESCRIPTION
**Pull request by anthonykoutroulis**
Not tested with flash, tested working by commenting out flash calls in seal_CTRL:150 and seal_CTRL:190.

Heap may need adjustment when activating flash code.

Tested working over USB

### CODE REVIEW

File: SealHATv1.0.4/tasks/seal_GPS.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ * seal_GPS.c
+ *
+ * Created: 5/11/2018 2:36:10 PM
+ *  Author: Anthony Koutroulis
+ */ 
+
+ #include "seal_GPS.h"
+
+TaskHandle_t        xGPS_th;    /* GPS task handle */
+GPS_MSG_t			gps_msg;	/* holds the GPS message to store in flash  */
```
Comment (TheSlat) [inline SealHATv1.0.4/tasks/seal_GPS.c]: Garbage! make it static and put inside the task so it isn't global.

File: SealHATv1.0.4/tasks/seal_GPS.c [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                    // TODO - snoop the logs and do something if consistently invalid
+                    gps_msg.header.size = logcount * sizeof(gps_log_t);
+                    err = ctrlLog_write((uint8_t*)&gps_msg, sizeof(DATA_HEADER_t) + gps_msg.header.size);
+                    if (err < 0) { 
+						gpio_toggle_pin_level(LED_RED); 
+					}
+                }
+            }
+        } else { /* the interrupt timed out */
+            // TODO - implement FIFO ready count (in gps.c) and do something with that
```
Comment (TheSlat) [inline SealHATv1.0.4/tasks/seal_GPS.c]: not sure what this TODO means, but this error condition should be logged to flash

File: SealHATv1.0.4/tasks/seal_GPS.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+ #include "seal_CTRL.h"
+ #include "sam-m8q/gps.h"
+
+#ifndef SEAL_GPS_H_
+#define SEAL_GPS_H_
+
+#define GPS_STACK_SIZE  (2048 / sizeof(portSTACK_TYPE))	/* number of 32-bit words to reserve for task */
+#define GPS_TASK_PRI    (tskIDLE_PRIORITY + 3)
+
+typedef enum GPS_NOTIFY_VALS {
```
Comment (TheSlat) [inline SealHATv1.0.4/tasks/seal_GPS.h]: look at the freeRTOS docs on directToTask, if you are only using one bit there is a MUCH better API to use.

File: SealHATv1.0.4/tasks/seal_GPS.c (hunk 1 above)
Comment (anthonykoutroulis) [reply SealHATv1.0.4/tasks/seal_GPS.c]: Done!

File: SealHATv1.0.4/tasks/seal_GPS.c (hunk 2 above)
Comment (anthonykoutroulis) [reply SealHATv1.0.4/tasks/seal_GPS.c]: This TODO refers to adding a Feature to check signal strength and/or put the device to sleep if X consecutive fixes are not "valid" (poor signal quality). I'm making this a feature issue.

Also, more logging for other possible errors were implemented.

**Pull request merged**